src/datachain/query/dataset.py

```diff
@@ -1778,6 +1778,10 @@ def query_wrapper(dataset_query: DatasetQuery) -> DatasetQuery:
     save = bool(os.getenv("DATACHAIN_QUERY_SAVE"))
     save_as = os.getenv("DATACHAIN_QUERY_SAVE_AS")
 
+    is_session_dataset = dataset_query.name.startswith(
+        Session.DATASET_PREFIX + dataset_query.session.name + "_"
+    )
```

Comment:
```suggestion
    is_session_dataset = dataset_query.name is not None and dataset_query.name.startswith(
        Session.DATASET_PREFIX + dataset_query.session.name + "_"
    )
```

`dataset_query` could be detached, so this needs a guard.